### PR TITLE
 Help command for makefile

### PR DESCRIPTION
**Pull request by suujia**
:tickets: **Ticket(s)**: Closes 🙆🏻‍♀️

I always forget all the commands and have to go to the file to look them up

---

## :construction_worker: Changes

command to reveal commands

## :flashlight: Testing Instructions

How convenient is this: 

```
$make help

Choose a command run in pinpoint:
  all                Runs check
  check              Run simple checks
  deps               Install dependencies
  test               Execute tests
  testenv            Set up test environment
  testenv-stop       Stop test environment
  monitoring         Set up monitoring environment
  monitoring-stop    Stop monitoring environment
  clean              Clean up stuff
  lint               Run linters and checks
  gen                Regenerate all generated code
  proto              Generate protobuf code from definitions
  proto-pkg          Runs protoc definitions 
  mocks              Generate database interface and mock
  core               Runs core service
  gateway            Runs API gateway
  gateway-tls        Runs gateway tls 
  web                Runs web app
  pinpoint-core      Builds binary for pinpoint-core
  pinpoint-gateway   Builds binary for pinpoint-gateway
```

### CODE REVIEW

**Review by bobheadxi — DISMISSED** [review]
# excuse me what the fuck? this is amazing?

File: Makefile [hunk 1]
```diff
@@ -99,41 +102,46 @@ mocks:
...
     ./core $(FLAGS)
 
-# Builds binary for pinpoint-gateway
+## pinpoint-gateway: Builds binary for pinpoint-gateway
 .PHONY: pinpoint-gateway
 pinpoint-gateway:
 	go build -o ./bin/pinpoint-gateway \
     -ldflags "-X main.Version=$(VERSION)" \
     ./gateway $(FLAGS)
+
```
Comment (bobheadxi) [inline Makefile]:
```suggestion
.PHONY: help
```

we don't have a file called `help`, but... just in case

**Review by bwdmonkey — DISMISSED** [review]
love it

File: Makefile [hunk 2]
```diff
@@ -2,16 +2,18 @@ VERSION=`git rev-parse --short HEAD`
 DEV_ENV=export `less ./dev/.env | xargs`
 TEST_COMPOSE=docker-compose -f dev/testenv.yml -p test
 MON_COMPOSE=docker-compose -f dev/monitoring.yml -p monitoring
+PROJECTNAME=$(shell basename "$(PWD)")
```
Comment (bobheadxi) [inline Makefile]: is there a particular reason this can't just be `pinpoint`?

Comment (suujia) [reply Makefile]: worst case scenario: project name changes!! :O frequently!!

**Review by bobheadxi — APPROVED** [review]

**Pull request merged**